utils/yoke/design_doc.md

```diff
@@ -1,20 +1,18 @@
-# Yoke: Lifetime Erasure for Rust
+# Yoke: Self-Referential Borrowing for Rust
 
 ## Problem statement
 
-Zero-copy deserialization is a very effective way to speed up programs and avoid allocations, however can lead to lifetimes pervasively spreading throughout the codebase, and prevents using diverse memory management techniques like caching.
+Zero-copy deserialization is a very effective way to speed up programs and avoid allocations. However, the requirement that data is borrowed from somewhere else means that:
 
-It would be nice if it were possible to "erase" lifetimes and turn them into dynamically managed lifetimes (similar to type erasure with `dyn`) to allow for more flexible memory management.
+1. All data types that contain zero-copy data, even indirectly, need to carry a lifetime parameter
+2. Certain memory management techniques are hampered, like caching.
+
+The goal of Yoke is to allow the borrowing of data from self, so that we don't need lifetime parameters to track data ownership, and to enable reference-counted data that can be safely dropped from a cache.
```

Comment:
ditto here; I'm fine with mentioning self-borrowing but I really think we should rely on the lifetime erasure analogy

Reply:
How's this?